src/test/kotlin/com/example/Employee/controllers/EmployeeControllerTest.kt

```diff
@@ -0,0 +1,62 @@
+package com.example.Employee.controllers
+
+import com.example.Employee.models.Employee
+import com.example.Employee.services.EmployeeService
+import io.mockk.every
+import io.mockk.mockk
+import io.mockk.verify
+import org.junit.jupiter.api.Test
+import org.amshove.kluent.`should be equal to`
+
+internal class EmployeeControllerTest{
+    private val employeeService=mockk<EmployeeService>()
+    private val employeeController=EmployeeController(employeeService)
+
+    @Test
+    fun `should create employee`() {
+        every { employeeService.addDetails(fakeEmployee) }returns Unit
```

Comment:
I didn't catch this bug earlier. On creation of the employee, how will the client (who makes the request), know the id of the employee entity that has just been created?

Reply:
The controller is returning the Employee object. Th client can id from there.

Reply:
Controller is not returning the object created by the server:
https://github.com/Sath200/Employee/blob/master/src/main/kotlin/com/example/Employee/controllers/EmployeeController.kt#L28

Client will pass the employee object in the request body with id = 0, or may not even pass the id.
This code [here](https://github.com/Sath200/Employee/blob/master/src/main/kotlin/com/example/Employee/controllers/EmployeeController.kt#L28) just returns back what the client sent. Not what the server created. If the server created an entity with id `53`, the client will not receive it.

Reply:
But I'm getting the id here. Please check this postman screenshot:
<img width="1440" alt="Screenshot 2022-05-27 at 5 01 03 PM" src="https://user-images.githubusercontent.com/102017428/170692397-ca08a2ef-0e06-4aea-8513-b39ef34b0d2f.png">

Reply:
I realise that it works, because of how `CrudRepository.save` modifies our `employee` object.
This is not advise-able, from an immutability perspective.
Also, look at this documentation of the `save` method:
![image](https://user-images.githubusercontent.com/2430621/170696962-3aebfc54-3ea6-41f3-b6cf-32e37d16fefe.png)

They advice us to use the return type.

Reply:
Okay .
I'll have to change this for all the services. Will do it.